core/src/main/java/org/infinispan/configuration/parsing/Parser80.java

```diff
@@ -2311,35 +2311,61 @@ private void parseIndexing(final XMLExtendedStreamReader reader, final Configura
                throw ParseUtils.unexpectedAttribute(reader, i);
          }
       }
-      Properties indexingProperties = parseProperties(reader);
+
+      Properties indexingProperties = new Properties();
+      while (reader.hasNext() && (reader.nextTag() != XMLStreamConstants.END_ELEMENT)) {
+         Element element = Element.forName(reader.getLocalName());
+         switch (element) {
+            case INDEXED_ENTITY: {
```

Comment:
I hated removing this handy type autodetection too, and resisted for years, but the Rome meeting convinced me there are greater benefits.

ATM by enabling indexing and not declaring any indexed entities you still get the old autodetection behaviour. If you declare at least one type autodetection is disabled and the list of types is locked. Putting undeclared entities in cache logs warnings while in autodetect mode and adds them to the list. And is completely ignored if autodetect is off. Everything would still be queryable anyway.